ethers-core/src/types/chain.rs

```diff
@@ -71,6 +57,32 @@ impl From<Chain> for u64 {
     }
 }
 
+impl TryFrom<u64> for Chain {
+    type Error = ();
+
+    fn try_from(chainid: u64) -> Result<Chain, Self::Error> {
```

Comment:
we could use a crate like https://crates.io/crates/num_enum to avoid duplicating the mapping here

Reply:
i think it's fine as-is, if we see the maintenance of this becoming a problem we can consider